File: 백준/Bronze/2798. 블랙잭/블랙잭.java

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    static int n, m;
    static int[] card;
    static int result = Integer.MIN_VALUE;
    public static void main(String[] args) throws Exception{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;

        st = new StringTokenizer(br.readLine());

        n = Integer.parseInt(st.nextToken());
        m = Integer.parseInt(st.nextToken());

        card = new int[n];

        st = new StringTokenizer(br.readLine());

        for (int i = 0; i < n; i++) {
            card[i] = Integer.parseInt(st.nextToken());
        }

        for (int i = 0; i < n - 2; i++) {
            for (int j = i + 1; j < n - 1; j++) {
                for (int k = j + 1; k < n; k++) {
                    result = card[i] + card[j] + card[k] <= m ? Math.max(card[i] + card[j] + card[k], result) : result;
                }
            }
        }
        System.out.println(result);

    }

}

File: 백준/Bronze/2798. 블랙잭/블랙잭.py

sentence = input()
sentence = sentence.split(' ')
N = int(sentence[0])
M = int(sentence[1])
card = input()
card = card.split(' ')

for i in range(N):
    card[i] = int(card[i])
result=set()
for i in range(N-2):
    for j in range(i+1,N-1):
        for k in range(j+1,N):
            val=card[i]+card[j]+card[k]
            if val<=M:
                result.add(val)

print(max(result))